src/common/key/graph.js

```diff
@@ -0,0 +1,29 @@
+import { isRange, isBranch }
+
+/*
+  Node types:
+
+  g - Graph Value
+  G - Graph Branch
+  r - Graph Range with start and end
+  s - Graph Range with start only
+  t - Graph Range with end only
+  u - Graph Range with neither start nor end
+  l - Graph Link
+
+  q - Query Leaf
+  Q - Query Branch
+  p - Query Range with Leaf
+  P - Query Range with Branch
+*/
+
+
+function encodeNode() {
+  if (isRange)
```

Comment:
Yes! I will remove it from this PR. The purpose of the "graph" encoder is quite different, it is just to have a more compact way of representing queries and results than using JSON. It is not really needed for functionality (just perf) and can be deferred.

The "key" encoder on the other hand is necessary for making indexes work correctly. JSON doesn't preserve order.

`JSON.stringify(10)` sorts before `JSON.stringify(2)`, for instance. If you have objects, the same object might have different JSON encodings because of key iteration order.